packages/vscode-extension/src/common/Project.ts

```diff
@@ -98,12 +102,20 @@ export const StartupStageWeight = [
 ];
 
 export type ReloadAction =
+  | "autoReload" // automatic reload mode
+  | "clearMetro" // clear metro cache, boot device, install app
   | "rebuild" // clean build, boot device, install app
   | "reboot" // reboots device, launch app
   | "reinstall" // force reinstall app
   | "restartProcess" // relaunch app
   | "reloadJs"; // refetch JS scripts from metro
 
+export enum ReloadResult {
+  succeeded,
+  failed,
+  canceled,
+}
```

Comment:
Can't we just rely on the exceptions to handle non-successful cases? Specifically if cancel uses a custom error type this should be cleaner IMO